src/test/java/com/devonfw/ide/sonarqube/common/impl/config/ConfigurationMapperTest.java

```diff
@@ -209,4 +213,28 @@ public void testFromJsonIllegalUndefinedDependency() {
         .containsExactly("Component 'component1' has dependency 'component2' but no such component is defined.");
   }
 
+  @Test
+  public void testFromJsonFileWhenPackagesIsPresent() {
+
+    // given
+    ConfigurationMapper mapper = new ConfigurationMapper();
+    File file = new File("src/test/architecture.json");
+    String expectedPattern = "([a-zA-Z0-9_]+\\.)+(persistence|core|service|gui)\\.([a-zA-Z0-9_]+)\\.(api|base|impl)(\\.[a-zA-Z0-9_]+)*";
+    List<String> expectedDetails = Arrays.asList("root", "layer", "component", "scope", "detail");
+    Map<String, String> expectedMappings = new HashMap<>();
+    expectedMappings.put("persistence", "dataaccess");
+    expectedMappings.put("core", "logic");
+    expectedMappings.put("gui", "client");
+
+    // when
+    Configuration config = mapper.fromJson(file);
+
+    // then
+    Packages packages = config.getArchitecture().getPackages();
+    assertThat(packages).isNotNull();
+    assertThat(packages.getPattern()).isEqualTo(expectedPattern);
+    assertThat(packages.getGroups()).isEqualTo(expectedDetails);
+    assertThat(packages.getMappings()).isEqualTo(expectedMappings);
```

Comment:
To make use of AssertJ and avoid relying on external `equals` implementation of different `Map`s, it would be more explicity to assert like this:
```
assertThat(packages.getMappings()).hasSize(3).containsEntry("persistence", "dataaccess").containsEntry("core", "logic").containsEntry("gui", "client");
```

Reply:
yes, sure, corrected.